File: the-lambda-power-tuner/typescript/lib/the-lambda-power-tuner-stack.ts
import * as cdk from '@aws-cdk/core';
import sam = require('@aws-cdk/aws-sam');
import lambda = require('@aws-cdk/aws-lambda');

export class TheLambdaPowerTunerStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    let powerValues = '128,256,512,1024,1536,3008';
    let lambdaResource = "*";

    // A lambda function to use to test the powertuner
    let exampleLambda = new lambda.Function(this, 'lambdaHandler', {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromInline('exports.handler = function(event, ctx, cb) { return cb(null, "hi"); }'),
      handler: 'index.handler'
    });

    // Uncomment to only allow this power tuner to manipulate this defined function
    //lambdaResource = exampleLambda.functionArn;

    // Output the Lambda function ARN in the deploy logs to ease testing
    new cdk.CfnOutput(this, 'LambdaARN', {
      value: exampleLambda.functionArn
    })

    // Deploy the aws-lambda-powertuning application from the Serverless Application Repository
    new sam.CfnApplication(this, 'powerTuner', {
      location: {
        applicationId: 'arn:aws:serverlessrepo:us-east-1:451282441545:applications/aws-lambda-power-tuning',
        semanticVersion: '3.3.3'
      },
      parameters: {
        "lambdaResource": lambdaResource,
        "PowerValues": powerValues
      }
    })
  }
}

File: the-lambda-power-tuner/java/src/main/java/com/cdkpatterns/TheLambdaPowerTunerStack.java
package com.cdkpatterns;

import java.util.Map;

import software.amazon.awscdk.core.CfnOutput;
import software.amazon.awscdk.core.Construct;
import software.amazon.awscdk.core.Stack;
import software.amazon.awscdk.core.StackProps;
import software.amazon.awscdk.services.lambda.Function;
import software.amazon.awscdk.services.lambda.Code;
import software.amazon.awscdk.services.lambda.Runtime;
import software.amazon.awscdk.services.sam.CfnApplication;
import software.amazon.awscdk.services.sam.CfnApplication.ApplicationLocationProperty;

public class TheLambdaPowerTunerStack extends Stack {
    public TheLambdaPowerTunerStack(final Construct scope, final String id) {
        this(scope, id, null);
    }
    
    private String powerValues = "128,256,512,1024,1536,3008";
    private String lambdaResource = "*";

    public TheLambdaPowerTunerStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        
        //  An example lambda that can be used to test the powertuner
        Function functionExample = Function.Builder.create(this, "exampleLambda")
        		.runtime(Runtime.NODEJS_12_X)
        		.handler("index.handler")
        		.code(Code.fromInline("exports.handler = function(event, ctx, cb) { return cb(null, 'hi'); }"))
        		.build();
        
        // uncomment to only allow this power tuner to manipulate this defined function
        // String lambdaResource = functionExample.getFunctionArn();

        // Output the Lambda function ARN in the deploy logs to ease testing
        CfnOutput.Builder.create(this, "LambdaARN")
        	.value(functionExample.getFunctionArn())
        	.build();
        
        // Deploy the aws-lambda-powertuning application from the Serverless Application Repository
        // https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:451282441545:applications~aws-lambda-power-tuning
        CfnApplication.Builder.create(this, "powerTuner")
        .location(new ApplicationLocationProperty
        		.Builder()
        		.applicationId("arn:aws:serverlessrepo:us-east-1:451282441545:applications/aws-lambda-power-tuning")
        		.semanticVersion("3.4.0")
        		.build())
        .parameters(Map.of(
        		"lambdaResource", lambdaResource,
        		"PowerValues", powerValues
        		))
        .build();
        
    }
}
